pkg/controller/traceflow/controller.go

```diff
@@ -224,14 +241,28 @@ func (c *Controller) checkTraceflowStatus(tf *opsv1alpha1.Traceflow) (retry bool
 	retry = false
 	sender := false
 	receiver := false
-	for _, nodeResult := range tf.Status.Results {
-		for _, ob := range nodeResult.Observations {
+	for i, nodeResult := range tf.Status.Results {
+		for j, ob := range nodeResult.Observations {
 			if ob.Component == opsv1alpha1.SpoofGuard {
 				sender = true
 			}
 			if ob.Action == opsv1alpha1.Delivered || ob.Action == opsv1alpha1.Dropped {
 				receiver = true
 			}
+			if ob.TranslatedDstIP != "" {
```

Comment:
Humm.. Feel not certain if worthwhile to figure out the Pod from IP. Do you think good enough to use IP to indicate the destination here?

Reply:
Yes, if a translated IP is a Pod IP, it can be retrieved here.
ob.TranslatedDstIP will exist only when packet is processed by DNAT.

Reply:
I meant there will be costs to watch Pods and build IP -> Pod indexes. So, I feel maybe not worthwhile to translate IP back to Pods.

Reply:
Pods will be always watched in AntreaController (by NetworkPolicyController), thus the only extra cost is the PodIP index. I feel an additional index is acceptable.

Reply:
Could you share the memory overhead of the new index your saw?
BTW, I know for networkpolicy query, I remember we plan to add Pod IP index too, but still want to avoid obvious mem increase. We already saw some mem issues in scale tests.

Reply:
@jianjuns for "antctl query", the index is from Pod Name to AddressGroup / AppliedToGroup. Memory overhead seemed to be at most 20%, at least in the tests we ran. Do you want us to add a configuration parameter, so that antctl query support can be disabled, in which case we would not build the index?

@jakesokol1

Reply:
In a 82 Pods cluster, the memory usage is 49.9M v.s. 53.6M (w/ PodIP indexer for Pods).
Do we have more accurate method to measure the memory usage? I cannot locate the Indexer memory usage in "pprof".

Reply:
Maybe you could consider writing a benchmark like Quan did for the NP controller: https://github.com/vmware-tanzu/antrea/blob/master/pkg/controller/networkpolicy/networkpolicy_controller_perf_test.go? You can then compare memory usage with / without the extra indexer in a cluster with 1000s of Pods.

Reply:
20% sounds not small. Believe @tnqn is looking at the scale issues. @tnqn : do you think it will make the mem issues we saw worse?

@antoninbas : do you think we can put "antctl query" under a single feature gate, or you believe we can just disable features require Pod name index? We can discuss more over the "antctl query" PR/issue too.

Reply:
Added 40K Pods to cache and verify 
```
v1.Pod{
	ObjectMeta: metav1.ObjectMeta{Namespace: namespace, Name: "pod1", UID: types.UID(uuid.New().String()), Labels: map[string]string{"app-1": "scale-1"}},
	Spec:       v1.PodSpec{NodeName: getRandomNodeName()},
	Status:     v1.PodStatus{PodIP: getRandomIP()},
}
```
The process memory usage is 131.3M v.s. 137.7M (w/ PodIP indexer for Pods).
6400K memory for 40K Pods. Looks like the overhead is not bad.

Reply:
131.3M v.s. 137.7M sounds ok. But in our scale tests, we saw mem usage of ~10G with 150K Pods and 40K NetworkPolicies. Do you think in that case, the extra index will lead to Gs of more mem consumption?

Reply:
If you can send me details of the specific case which worries you, I can add an additional test to networkpolicy_controller_perf_test/controller_query_perf_test to directly test memory consumption.

Reply:
> 131.3M v.s. 137.7M sounds ok. But in our scale tests, we saw mem usage of ~10G with 150K Pods and 40K NetworkPolicies. Do you think in that case, the extra index will lead to Gs of more mem consumption?

I think the major different point is that I only added many Pods into cache and didn't touch NetworkPolicy. For NetworkPolicy, many additional elements will be cached such as rules and OVS flows.

Reply:
The price of the extra PodIP index is a map like below will be created and maintained when adding/updating/deleting pods. But unlike the Pod index of AddressGroup and AppliedToGroup that could have large number of Pods, each Pod has only 1 IP, the extra overhead is constant.
```
{
    "1.1.1.1": []string{"ns1/pod1"},
    "1.1.1.2": []string{"ns1/pod2"},
}
```
```
BenchmarkAddPortWithPodIPIndex-4          252216              4207 ns/op            1608 B/op         17 allocs/op
BenchmarkAddPortWithoutPodIPIndex-4       408417              3234 ns/op            1144 B/op         12 allocs/op
```
Given the difference is 3us vs 4us per pod, i.e. 300ms vs 400ms for 100k Pods, I think it should be ok.

Reply:
Ok

Reply:
Added comments to:
pkg/controller/traceflow/controller.go
```
	// Add IP-Pod index. Each Pod has only 1 IP, the extra overhead is constant and acceptable.
	// @tnqn evaluated the performance without/with IP index is 3us vs 4us per pod, i.e. 300ms vs 400ms for 100k Pods.
	podInformer.Informer().AddIndexers(cache.Indexers{podIPIndex: podIPIndexFunc})
```